### PR TITLE
feat(apig): add datasource to get basic configurations of the api

### PR DESCRIPTION
**Pull request by wuzhuanhong**
**What this PR does / why we need it**:

Add new datasource to get basci configurations of the api under specified dedicated  APIG instance.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
 1. add new dataSource to get basci configurations of the api.
 2. add corresponding to document and acceptance test.

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/services/acceptance/apig TESTARGS='-run TestAccDataSourceApiBasicConfigurations_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run TestAccDataSourceApiBasicConfigurations_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceApiBasicConfigurations_basic
=== PAUSE TestAccDataSourceApiBasicConfigurations_basic
=== CONT  TestAccDataSourceApiBasicConfigurations_basic
--- PASS: TestAccDataSourceApiBasicConfigurations_basic (680.19s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      680.234s
```

### CODE REVIEW

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/apig TESTARGS='-run TestAccDataSourceApiBasicConfigurations_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run TestAccDataSourceApiBasicConfigurations_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceApiBasicConfigurations_basic
=== PAUSE TestAccDataSourceApiBasicConfigurations_basic
=== CONT  TestAccDataSourceApiBasicConfigurations_basic
--- PASS: TestAccDataSourceApiBasicConfigurations_basic (692.73s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      692.795s

File: docs/data-sources/apig_api_basic_configurations.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+* `request_protocol` - (Optional, String) Specifies the request protocol of the API.  
+  The valid values are `HTTP`, `HTTPS`, `BOTH` and `GRPCS`.
+
+* `security_authentication` - (Optional, String) Specifies the security authentication mode of the API request.  
+  The valid values are `NONE`, `APP`, `IAM` and `AUTHORIZER`.
+
+* `vpc_channel_name` - (Optional, String) Specifies the name of the VPC channel. Fuzzy search is supported.
+
+* `precise_search` - (Optional, String) Specifies the parameter name for exact matching.  
+  The valid value is `name` or `req_uri`.
```
Comment (Lance52259) [inline docs/data-sources/apig_api_basic_configurations.md]:
```suggestion
* `precise_search` - (Optional, String) Specifies the parameter name that you want to match exactly.
  The valid values are:
  + **name**: API name, corresponding to the field `name` in this data source arguments.
  + **req_uri**: Request path, corresponding to the field `request_path` in this data source arguments.

  This parameter takes effect only after the corresponding parameter(s) is(are) set.
```
And check whether this parameter can be set to multiple enumerated values, e.g. `name,req_uri`.

File: docs/data-sources/apig_api_basic_configurations.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+* `authorizer_id` - The ID of the authorizer to which the API request used.
+
+* `publish_id` - The ID of publish corresponding to the API.
+
+* `backend_type` - The backend type of the API.
+
+* `cors` - Whether CORS is supported.
+
+* `matching` - The matching mode of the API.  
+  The valid value is `Exact` or `Prefix`.
```
Comment (Lance52259) [inline docs/data-sources/apig_api_basic_configurations.md]: If you want to list the enumerated values of the property, write as:
```suggestion
* `matching` - The matching mode of the API.  
  + **Exact**
  + **Prefix**
```

File: huaweicloud/services/acceptance/apig/data_source_huaweicloud_apig_api_basic_configurations_test.go [hunk 3]
```diff
@@ -0,0 +1,436 @@
...
+  id_filter_result = [
+    for v in data.huaweicloud_apig_api_basic_configurations.filter_by_id.configurations[*].id : v == local.api_id
+  ]
+}
+
+output "is_id_filter_useful" {
+  value = length(local.id_filter_result) > 0 && alltrue(local.id_filter_result)
+}
+
+# Filter by name (fuzzy search)
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/apig/data_source_huaweicloud_apig_api_basic_configurations_test.go:213]: please test whether the corresponding parameter (name) supports exact match.

File: docs/data-sources/apig_api_basic_configurations.md (hunk 1 above)
Comment (wuzhuanhong) [reply docs/data-sources/apig_api_basic_configurations.md]: fixed

File: docs/data-sources/apig_api_basic_configurations.md (hunk 2 above)
Comment (wuzhuanhong) [reply docs/data-sources/apig_api_basic_configurations.md]: fixed

File: huaweicloud/services/acceptance/apig/data_source_huaweicloud_apig_api_basic_configurations_test.go (hunk 3 above)
Comment (wuzhuanhong) [reply huaweicloud/services/acceptance/apig/data_source_huaweicloud_apig_api_basic_configurations_test.go:213]: fixed

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/apig TESTARGS='-run TestAccDataSourceApiBasicConfigurations_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run TestAccDataSourceApiBasicConfigurations_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceApiBasicConfigurations_basic
=== PAUSE TestAccDataSourceApiBasicConfigurations_basic
=== CONT  TestAccDataSourceApiBasicConfigurations_basic
--- PASS: TestAccDataSourceApiBasicConfigurations_basic (735.06s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      735.131s

**Pull request merged**